libfreerdp/core/gateway/rts.c

```diff
@@ -887,7 +887,7 @@ int rts_recv_flow_control_ack_pdu(rdpRpc* rpc, BYTE* buffer, UINT32 length)
 	BYTE ChannelCookie[16];
 
 	offset = 24;
-	offset += rts_flow_control_ack_command_read(rpc, &buffer[offset], length - offset,
```

Comment:
@hardening the thing is that even if we start commenting out everything, it defeats the purpose. This only becomes annoying because of the clang checker warnings, which only catches certain cases. For instance, our Stream macros update pointers of the wStream* struct on every "call" regardless if they are actually required later on. The last call to a read operation that updates the read pointer will in many cases be discarded unless we do a check for the end read position, but it doesn't mean we should systematically get rid of them. One other way to get rid of many of the warnings would be to define data structures for each of the messages and refactor the code to have serialization/deserialization functions separate from the code acting on the packet contents. Functions that read and populate a data structure do not generate warnings, as there is no way to know if the data populated in the data structure will be used or not in the future. Isn't there a way to tweak the clang warnings?